apps/extension/src/core/managers/config.ts

```diff
@@ -232,10 +235,21 @@ class ConfigManager extends BaseManager<Config> {
   }
 
   getCaller(config: Config) {
-    return this.getCallerForAuth(config.auth, this.getCurrentModel(config))
+    return this.getCallerForAuth(config.auth, this.sync_getModel(config))
   }
 
-  getCurrentModel(config: Config): ModelID | undefined {
+  async predictModel(
+    config: Config,
+    txn?: Transaction
+  ): Promise<ModelID | string> {
+    const currentModel = this.sync_getModel(config)
+    if (currentModel) {
+      return currentModel
+    }
+    return unwrap(await modelRouter.route(config, txn))
+  }
+
+  sync_getModel(config: Config): ModelID | undefined {
```

Comment:
is the idea to have an async version of this down the line?

Reply:
Should have renamed this actually, since `predictModel` used to be the async version. will change to getModel